lib/fastlane/plugin/sentry/actions/sentry_upload_mobile_app.rb

```diff
@@ -0,0 +1,69 @@
+module Fastlane
+  module Actions
+    class SentryUploadMobileAppAction < Action
+      def self.run(params)
+        require 'shellwords'
+
+        Helper::SentryConfig.parse_api_params(params)
+
+        # Verify xcarchive path
+        xcarchive_path = params[:xcarchive_path]
+        UI.user_error!("Could not find xcarchive at path '#{xcarchive_path}'") unless File.exist?(xcarchive_path)
+        UI.user_error!("Path '#{xcarchive_path}' is not an xcarchive") unless File.extname(xcarchive_path) == '.xcarchive'
+
+        command = [
+          "mobile-app",
+          "upload",
+          File.absolute_path(xcarchive_path)
+        ]
+
+        Helper::SentryHelper.call_sentry_cli(params, command)
+        UI.success("Successfully uploaded mobile app archive: #{xcarchive_path}")
+      end
+
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.description
+        "Upload iOS app archive to Sentry"
+      end
+
+      def self.details
+        [
+          "This action allows you to upload iOS app archives (.xcarchive) to Sentry.",
+          "The uploaded archive can be used for various Sentry features including improved symbolication and source context."
+        ].join(" ")
+      end
+
+      def self.available_options
+        Helper::SentryConfig.common_api_config_items + [
+          FastlaneCore::ConfigItem.new(key: :xcarchive_path,
+                                       env_name: "SENTRY_XCARCHIVE_PATH",
```

Comment:
Similarly I'm not sure we need an env var for this if we are re-using the existing lane stuff.